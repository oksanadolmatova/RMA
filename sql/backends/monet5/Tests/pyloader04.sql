
# unmatched element (a3) present in dict, this should throw an error
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({'a1': 3, 'a2': 4, 'a3': 5})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# use non-string type as key
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({'a1': 3, 'a2': 4, 3: 5})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# return empty list
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({'a1': [], 'a2': numpy.array([])})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# empty dictionary
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# unsupported python object
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	class MyClass:
		i = 1234

	_emit.emit({'a1': MyClass()})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# fail str -> int conversion
START TRANSACTION;
CREATE TABLE pyloader04table(a1 int, a2 int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({'a1': 'hello'})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
ROLLBACK;

# test quoted names
START TRANSACTION;
CREATE TABLE pyloader04table("select" int, "from" int);
CREATE LOADER pyloader04() LANGUAGE PYTHON {
	_emit.emit({'select': 3, 'from': 4})
};
COPY LOADER INTO pyloader04table FROM pyloader04();
SELECT * FROM pyloader04table;
DROP TABLE pyloader04table;
DROP LOADER pyloader04;
ROLLBACK;
