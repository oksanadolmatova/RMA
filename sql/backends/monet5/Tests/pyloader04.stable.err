stderr of test 'pyloader04` in directory 'sql/backends/monet5` itself:


# 12:30:43 >  
# 12:30:43 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34261" "--set" "mapi_usock=/var/tmp/mtest-7433/.s.monetdb.34261" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/mytherin/opt/var/MonetDB/mTests_sql_backends_monet5" "--set" "embedded_r=yes" "--set" "embedded_py=true"
# 12:30:43 >  

# builtin opt 	gdk_dbpath = /home/mytherin/opt/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 34261
# cmdline opt 	mapi_usock = /var/tmp/mtest-7433/.s.monetdb.34261
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/mytherin/opt/var/MonetDB/mTests_sql_backends_monet5
# cmdline opt 	embedded_r = yes
# cmdline opt 	embedded_py = true
# cmdline opt 	gdk_debug = 536870922

# 12:30:43 >  
# 12:30:43 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-7433" "--port=34261"
# 12:30:43 >  

MAPI  = (monetdb) /var/tmp/mtest-6147/.s.monetdb.38930
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !
        !  1. def pyfun(_emit,_conn):
        !> 2.   _emit.emit({'a1': 3, 'a2': 4, 'a3': 5})
        !  3. 
        !Unmatched element "a3" in dict
CODE  = PY000
MAPI  = (monetdb) /var/tmp/mtest-29559/.s.monetdb.38002
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !
        !  1. def pyfun(_emit,_conn):
        !> 2.   _emit.emit({'a1': 3, 'a2': 4, 3: 5})
        !  3. 
        !Unmatched element "3" in dict
CODE  = PY000
MAPI  = (monetdb) /var/tmp/mtest-29559/.s.monetdb.38002
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !
        !  1. def pyfun(_emit,_conn):
        !> 2.   _emit.emit({'a1': [], 'a2': numpy.array([])})
        !  3. 
        !Empty input values supplied
CODE  = PY000
MAPI  = (monetdb) /var/tmp/mtest-29559/.s.monetdb.38002
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !
        !  1. def pyfun(_emit,_conn):
        !> 2.   _emit.emit({})
        !  3. 
        !dict must contain at least one element
CODE  = PY000
MAPI  = (monetdb) /var/tmp/mtest-29559/.s.monetdb.38002
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !  2.   class MyClass:
        !  3.     i = 1234
        !> 4.   _emit.emit({'a1': MyClass()})
        !  5. 
        !Unsupported Python Object <type 'instance'>
CODE  = PY000
MAPI  = (monetdb) /var/tmp/mtest-29559/.s.monetdb.38002
QUERY = COPY LOADER INTO pyloader04table FROM pyloader04();
ERROR = !Python exception
        !
        !  1. def pyfun(_emit,_conn):
        !> 2.   _emit.emit({'a1': 'hello'})
        !  3. 
        !Failed conversion: Error converting string.
CODE  = PY000

# 12:30:44 >  
# 12:30:44 >  "Done."
# 12:30:44 >  

