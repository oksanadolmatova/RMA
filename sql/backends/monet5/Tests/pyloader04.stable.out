stdout of test 'pyloader04` in directory 'sql/backends/monet5` itself:


# 12:30:43 >  
# 12:30:43 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=34261" "--set" "mapi_usock=/var/tmp/mtest-7433/.s.monetdb.34261" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/mytherin/opt/var/MonetDB/mTests_sql_backends_monet5" "--set" "embedded_r=yes" "--set" "embedded_py=true"
# 12:30:43 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 7.682 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://mytherin:34261/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-7433/.s.monetdb.34261
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

Ready.
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 24_zorder.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql

# 12:30:43 >  
# 12:30:43 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-7433" "--port=34261"
# 12:30:43 >  

#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({'a1': 3, 'a2': 4, 'a3': 5})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({'a1': 3, 'a2': 4, 3: 5})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({'a1': [], 'a2': numpy.array([])})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	class MyClass:
#		i = 1234
#
#	_emit.emit({'a1': MyClass()})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table(a1 int, a2 int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({'a1': 'hello'})
#};
#ROLLBACK;
#START TRANSACTION;
#CREATE TABLE pyloader04table("select" int, "from" int);
#CREATE LOADER pyloader04() LANGUAGE PYTHON {
#	_emit.emit({'select': 3, 'from': 4})
#};
#COPY LOADER INTO pyloader04table FROM pyloader04();
#SELECT * FROM pyloader04table;
% sys.pyloader04table,	sys.pyloader04table # table_name
% select,	from # name
% int,	int # type
% 1,	1 # length
[ 3,	4	]
#DROP TABLE pyloader04table;
#DROP LOADER pyloader04;
#ROLLBACK;

# 12:30:44 >  
# 12:30:44 >  "Done."
# 12:30:44 >  

